core/include/traccc/edm/track_state.hpp

```diff
@@ -77,10 +77,23 @@ struct track_state {
                       "The measurement dimension should be 1 or 2");
 
         matrix_type<D, 1> ret;
-        matrix_operator().element(ret, 0, 0) = m_measurement.local[0];
-        if constexpr (D == 2u) {
-            matrix_operator().element(ret, 1, 0) = m_measurement.local[1];
+        if (m_measurement.subs.get_indices()[0] == e_bound_loc0) {
+            matrix_operator().element(ret, 0, 0) = m_measurement.local[0];
+            if constexpr (D == 2u) {
+                matrix_operator().element(ret, 1, 0) = m_measurement.local[1];
+            }
+        } else if (m_measurement.subs.get_indices()[0] == e_bound_loc1) {
+            matrix_operator().element(ret, 0, 0) = m_measurement.local[1];
+            if constexpr (D == 2u) {
+                matrix_operator().element(ret, 1, 0) = m_measurement.local[0];
+            }
+        } else {  // same as e_bound_loc0
```

Comment:
I think we don't address such case at the moment
Let us put the warning in the if block:
```c++
assert("The measurement index out of e_bound_loc0 and e_bound_loc1 should not happen.");
```